### PR TITLE
Feature/resumption data error handling

### LINKED ISSUE #2486: [SDL 0190] Handle response from HMI during resumption data
Proposal: [Handle response from HMI during resumption data](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0190-resumption-data-error-handling.md)

> The aim of this feature is to handle HMI error responses while restoring the data during resumption process.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/562

### PR DESCRIPTION
**Pull request by AByzhynar**
:exclamation:  **NOTE**  :exclamation:  :  **This PR should be merged BEFORE #2674 and #2740 as it is the basis for implementation of [Interior Vehicle Data resumption](https://github.com/smartdevicelink/sdl_evolution/issues/554) and [Button Subscription response from HMI](https://github.com/smartdevicelink/sdl_evolution/issues/568)**

Fixes #2486 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
ATF test scripts provided in [feature/resumption_data_error_handling](https://github.com/smartdevicelink/sdl_atf_test_scripts/tree/feature/resumption_data_error_handling)

### HMI implementation
HMI implementation provided in https://github.com/smartdevicelink/sdl_hmi/pull/114

### Summary
Implemented logic of HMI erroneous responses handling during resumption process

### Tasks Remaining:
- [ ] [Update unit tests for commands]

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/include/application_manager/application_manager_impl.h [hunk 1]
```diff
@@ -346,6 +346,10 @@ class ApplicationManagerImpl
    * @brief Closes all registered applications
    */
   void UnregisterAllApplications();
+
+  DEPRECATED bool RemoveAppDataFromHMI(ApplicationSharedPtr app);
```
Comment (yang1070) [inline src/components/application_manager/include/application_manager/application_manager_impl.h]: why DEPRECATED a new added function?

File: src/components/application_manager/include/application_manager/application_manager_impl.h [hunk 2]
```diff
@@ -346,6 +346,10 @@ class ApplicationManagerImpl
...
+
+  DEPRECATED bool LoadAppDataToHMI(ApplicationSharedPtr app);
```
Comment (yang1070) [inline src/components/application_manager/include/application_manager/application_manager_impl.h]: why DEPRECATED a new added function?

**Comment by dboltovskyi** [comment]
Rebuild required

**Comment by LuxoftAKutsan** [comment]
@mked-luxoft  please take care about commit history

**Comment by yang1070** [comment]
@AByzhynar Please rebase to the latest develop branch.

**Comment by ychernysheva** [comment]
Closed as outdated in favor of https://github.com/smartdevicelink/sdl_core/pull/3475

**Pull request closed without merging**